vm/src/org/mozartoz/truffle/nodes/local/BindVarValueNode.java

```diff
@@ -19,30 +20,47 @@ public static BindVarValueNode create() {
 
 	@Specialization(guards = "var.getNext() == var")
 	Object bind1(Variable var, Object value) {
+		print_nlinks(var);
 		var.setInternalValue(value, var);
 		return value;
 	}
 
 	@Specialization(guards = { "var.getNext() != var", "var.getNext().getNext() == var" })
 	Object bind2(Variable var, Object value) {
+		print_nlinks(var);
+		Variable next = var.getNext();
 		var.setInternalValue(value, var);
-		var.getNext().setInternalValue(value, var);
+		next.setInternalValue(value, var);
 		return value;
 	}
 
 	@Specialization(guards = { "var.getNext() != var", "var.getNext().getNext().getNext() == var" })
 	Object bind3(Variable var, Object value) {
+		print_nlinks(var);
+		Variable next1 = var.getNext(), next2 = next1.getNext();
```

Comment:
No multiple declarations on the same line.
Move them right before it's needed so we have pairs of getNext/setInternalValue.